src/plugins/pinDms/index.tsx

```diff
@@ -1,133 +1,368 @@
+import { DEFAULT_CHUNK_SIZE } from "./constants";
+import { canMoveCategory, canMoveCategoryInDirection, categories, Category, categoryLen, collapseCategory, getAllUncollapsedChannels, getSections, initCategories, isPinned, migrateData, moveCategory, removeCategory } from "./data";
 
-export default definePlugin({
-    name: "PinDMs",
-    description: "Allows you to pin private channels to the top of your DM list. To pin/unpin or reorder pins, right click DMs",
-    authors: [Devs.Ven, Devs.Strencher],
+interface ChannelComponentProps {
+    children: React.ReactNode,
+    channel: Channel,
+    selected: boolean;
+}
 
-    settings,
-    contextMenus,
+const headerClasses = findByPropsLazy("privateChannelsHeaderContainer");
 
-    usePinCount(channelIds: string[]) {
-        const pinnedDms = usePinnedDms();
-        // See comment on 2nd patch for reasoning
-        return channelIds.length ? [pinnedDms.size] : [];
-    },
+const PrivateChannelSortStore = findStoreLazy("PrivateChannelSortStore") as { getPrivateChannelIds: () => string[]; };
 
-    getChannel(channels: Record<string, Channel>, idx: number) {
-        return channels[getPinAt(idx)];
-    },
+export let instance: any;
+export const forceUpdate = () => instance?.props?._forceUpdate?.();
 
-    isPinned,
-    getSnapshot: sortedSnapshot,
+// the flux property in definePlugin doenst fire, probably because startAt isnt Init
+waitFor(["dispatch", "subscribe"], m => {
+    m.subscribe("CONNECTION_OPEN", async () => {
+        if (!Settings.plugins.PinDMs?.enabled) return;
 
-    getScrollOffset(channelId: string, rowHeight: number, padding: number, preRenderedChildren: number, originalOffset: number) {
-        if (!isPinned(channelId))
-            return (
-                (rowHeight + padding) * 2 // header
-                + rowHeight * snapshotArray.length // pins
-                + originalOffset // original pin offset minus pins
-            );
+        const id = UserStore.getCurrentUser()?.id;
+        await initCategories(id);
+        await migrateData(id);
+        forceUpdate();
+        // dont want to unsubscribe because if they switch accounts we want to reinit
+    });
+});
 
-        return rowHeight * (snapshotArray.indexOf(channelId) + preRenderedChildren) + padding;
+
+export const settings = definePluginSettings({
+    sortDmsByNewestMessage: {
+        type: OptionType.BOOLEAN,
+        description: "Sort DMs by newest message",
+        default: false,
+        onChange: () => forceUpdate()
     },
 
+    dmSectioncollapsed: {
+        type: OptionType.BOOLEAN,
+        description: "Collapse DM sections",
+        default: false,
+        onChange: () => forceUpdate()
+    }
+});
+
+export default definePlugin({
+    name: "PinDMs",
+    description: "Allows you to pin private channels to the top of your DM list. To pin/unpin or reorder pins, right click DMs",
+    authors: [Devs.Ven, Devs.Strencher, Devs.Aria],
```

Comment:
he wants himself removed
```suggestion
    authors: [Devs.Ven, Devs.Aria],
```